src/forms/output/ResourcesColumns.js

```diff
@@ -0,0 +1,183 @@
+import {
+  HighchartsChart,
+  Chart,
+  Title,
+  Legend,
+  Tooltip,
+  XAxis,
+  YAxis,
+  ColumnSeries,
+  withHighcharts,
+} from 'react-jsx-highcharts';
+
+// LOGIC
+import { timescaleMultipliers } from '../../utils/convert-by-timescale';
+import { getChartData } from '../../utils/charts/getChartData';
+import { toFancyMoneyStr } from '../../utils/charts/chartFormatting';
+import {
+  formatMoneyWithK,
+  snippetToText,
+} from './chartStringTransformers';
+
+
+// Graphs get things in monthly values, so we'll convert from there
+let multipliers = timescaleMultipliers.fromMonthly;
+
+
+/** Graph of all incoming resources as household income changes. Uses Highchart lib.
+ * @class
+ *
+ * @params {object} props
+ * @params {object} props.client Data for current and future client circumstances.
+ * @params {'Weekly'|'Monthly'|'Yearly'} props.timescale Interval of time to show for pay amount.
+ * @params {string[]} props.activePrograms Benefit programs in which the household enrolled.
+ * @params {object} props.className An extra class for the outermost component,
+ *     whether it's the chart or the no-chart message.
+ * @params {object} props.snippets Translation spans of text in app.
+ */
+class ResourcesColumnsComp extends Component {
+
+  constructor (props) {
+    super(props);
+    let separator = snippetToText(props.snippets.i_thousandsSeparator);
+    // This doesn't affect the strings we put in there, just pure numbers
+    Highcharts.setOptions({ lang: { thousandsSep: separator }});
+  }
+
+  render () {
+    const {
+      client,
+      timescale,
+      activePrograms,
+      className,
+      snippets,
+    } = this.props;
+
+    let clone = cloneDeep(client);
+
+    let multiplier    = multipliers[ timescale ],
+        resources     = [ `earned` ].concat(activePrograms),
+        currentEarned = client.current.earned * multiplier,
+        futureEarned  = client.future.earned * multiplier,
+        getText       = snippetToText;
+
+    // Adjust to time-interval. Highcharts will round
+    // for displayed ticks.
+    const xRange   = [
+            currentEarned,
+            futureEarned, 
+          ],  // x-axis/earned income numbers
+          datasets = getChartData(xRange, multiplier, clone, resources, {});
+
+    // Columns and categories for each pay amount
+    let columns    = [],
+        categories = [];
+    for (let dataseti = 0; dataseti < datasets.length; dataseti++) {
+      let dataset = datasets[ dataseti ],
+          column = (
+            <ColumnSeries
+              key         = { dataset.label }
+              id          = { dataset.label.replace(` `, ``) }
+              name        = { dataset.label }
+              data        = { dataset.data }
+              legendIndex = { dataseti }
+              x           = { dataseti }
+              label       = { dataseti } />
+          );
+
+      columns.unshift(column);
+
+      if (dataset.label === `Earned`) {
```

Comment:
This should probably use an ID or something; when this is localized, this will be a pain to use as the conditional.

Reply:
Good point. @todo.